### PR TITLE
Added lower case alphabet range to regex pattern

### LINKED ISSUE #115: Adding a string instead of real UUID's in giftaid-update url is going to a sorry page
Eg: https://giftaid-staging.comicrelief.com/update/test is going to a sorry page.

it accepts only real UUID's

https://dashboard.iopipe.com/project/65d0875a-e96a-4a1e-8a08-976790bf45d2/alert/1001a1af-ae07-445f-af4d-139d2db0f7d8/event/8644a375-dd0c-4541-8baf-18db465632ef

### LINKED ISSUE #140: Remove transaction id regex check
Remove `transactionId` field and url parameter pattern and regex validation.

### PR DESCRIPTION
**Pull request by s-barrah**
Fixes #115 - added lower case regex pattern
Fixes #140 - removes `transactionId` regex test

### CODE REVIEW

**Comment by KrupaPammi** [comment]
@s-barrah I've tested this on the netlify PR envt and working as expected. Could you please update any 1 nightwatch test to have the transaction ID in lower case so we cover that.

File: cypress/integration/update_spec.js [hunk 1]
```diff
@@ -37,16 +37,16 @@ describe('e2e test typing transaction ID and choosing "yes" to claim gift aid on
         cy.get('#field-input--transactionId').clear()
         cy.get('#field-input--firstname').click()
         cy.get('#field-error--transactionId > span').should('contain','Please fill in your transaction id')
-        cy.get('#field-input--transactionId').type('#£$^')
-        cy.get('#field-error--transactionId > span').should('contain','This does not match a transaction ID in our system, please check your donation confirmation email or letter')
         cy.get('#field-input--transactionId').clear().type('d-BEXd501')
-        cy.get('#field-error--transactionId > span').should('contain','This does not match a transaction ID in our system, please check your donation confirmation email or letter')
+        cy.get('#field-error--transactionId > span').should('be.not.visible')
```
Comment (Heleen-cr) [inline cypress/integration/update_spec.js]: I know the tests passed on the PR, but is this syntax really correct?
Thought it was `not.be.visible` or is cypress flexible like that?

**Comment by Heleen-cr** [comment]
Can I suggest adding the following pattern to the transactionId field in the defaultUpdateFields.json file?
`"pattern": "^[a-zA-Z0-9-]{5,}$"`
Instead of doing separate checks specifically for this field?
This should show the error message immediately instead of having to press the submit button first.

File: cypress/integration/update_spec.js (hunk 1 above)
Comment (s-barrah) [reply cypress/integration/update_spec.js]: I didn't really change anything here. `be.not.visible` is used in multiple lines, simply just duplicated the test.

**Comment by s-barrah** [comment]
> Can I suggest adding the following pattern to the transactionId field in the defaultUpdateFields.json file?
> `"pattern": "^[a-zA-Z0-9-]{5,}$"`
> (allow lower case characters, upper case characters, the numbers 0 to 9 and a dash from the beginning to the end of the string, but there have to be 5 or more characters)
> Instead of doing separate checks specifically for this field?
> This should show the error message immediately instead of having to press the submit button first.
> 
> The InputField component should nicely deal with the validation you see.

This makes sense. Thanks.

File: src/pages/UpdateForm/UpdateForm.js [hunk 2]
```diff
@@ -230,6 +230,21 @@ class UpdateForm extends Component {
...
+          validation: {
+            ...this.state.validation,
+            transactionId: {
+              valid: false,
+              value: childState.value,
+              message: childState.message,
+              showErrorMessage: true,
+            },
+          },
+        });
```
Comment (Heleen-cr) [inline src/pages/UpdateForm/UpdateForm.js]: is this still necessary?

Comment (s-barrah) [reply src/pages/UpdateForm/UpdateForm.js]: Nope.

**Pull request merged**